include/boost/geometry/algorithms/detail/calculate_point_order.hpp

```diff
@@ -297,7 +297,11 @@ struct calculate_point_order_by_area
                 Ring, Strategy
             >::type result_type;
 
-        result_type const result = ring_area_type::apply(ring, strategy);
+        result_type const result = ring_area_type::apply(ring,
+													     geometry::strategies::area::services::strategy_converter
+															<
+																decltype(strategy.get_area_strategy())
+															>::get(strategy.get_area_strategy()));
```

Comment:
Here is another place. Note `decltype` instead of getting the type from the strategy type.